# 1. Business Problem :

A vaccine for the H1N1 flu virus became publicly available in October 2009. In late 2009 and early 2010, the United States conducted the National 2009 H1N1 Flu Survey. This phone survey asked respondents whether they had received the H1N1 and seasonal flu vaccines, in conjunction with questions about themselves. These additional questions covered their social, economic, and demographic background, opinions on risks of illness and vaccine effectiveness, and behaviors towards mitigating transmission.

We are going to analyse the different data in order :
* to have a better understanding of how these characteristics are associated with personal vaccination patterns.
* to predict the personal vaccination patterns
* to provide guidance for future public health efforts.



# 2. Methodology :

### Step 1: Perform data acquisition, preparation & cleaning

- Load the dataset into a data frame using Pandas
- Explore the number of rows & columns, ranges of values etc.
- Handle missing, incorrect and invalid data
- Perform any additional steps (parsing dates, creating additional columns, merging multiple dataset etc.)

### Step 2: Perform exploratory analysis & visualization

- Compute the mean, sum, range and other interesting statistics for numeric columns
- Explore distributions of numeric columns using histograms etc.
- Explore relationship between columns using scatter plots, bar charts etc.
- Make a note of interesting insights from the exploratory analysis

### Step 3: Ask & answer questions about the data

- Ask at some interesting questions about our dataset
- Answer the questions either by computing the results using Numpy/Pandas or by plotting graphs using Matplotlib/Seaborn
- Create new columns, merge multiple dataset and perform grouping/aggregation wherever necessary
- Wherever we're using a library function from Pandas/Numpy/Matplotlib etc. explain briefly what it does

### Step 4: Create a model of the prediction and Evaluate them

- Create a CNN model and predict the labels
- Create a RNN model and predict the labels
- Create a GRU model and predict the labels
- We’ll use a series of machine/Deep learning evaluation metrics (ROC, AUC, sensitivity, specificity, etc) as well as Health-oriented metrics (herd immunity).


### Step 5: Summarize our inferences & write a conclusion

- Write a summary of what we've learned from the analysis
- Include interesting insights and graphs from previous sections
- Share ideas for future work on the same topic using other relevant datasets
- Share links to resources we found useful during our analysis

# 3. Acquisition, preparation of the Data

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

## 3.1. Load the dataset into a data frame using Pandas


In [2]:
# We need to load training and labels dataset to prepare them for the analysis / modelling
training_features = pd.read_csv("training_set_features.csv")
training_labels = pd.read_csv("training_set_labels.csv")

test_features = pd.read_csv("test_set_features.csv")


In [3]:
training_features.head()

respondent_id  h1n1_concern  h1n1_knowledge  behavioral_antiviral_meds  \
0              0           1.0             0.0                        0.0   
1              1           3.0             2.0                        0.0   
2              2           1.0             1.0                        0.0   
3              3           1.0             1.0                        0.0   
4              4           2.0             1.0                        0.0   

   behavioral_avoidance  behavioral_face_mask  behavioral_wash_hands  \
0                   0.0                   0.0                    0.0   
1                   1.0                   0.0                    1.0   
2                   1.0                   0.0                    0.0   
3                   1.0                   0.0                    1.0   
4                   1.0                   0.0                    1.0   

   behavioral_large_gatherings  behavioral_outside_home  \
0                          0.0                      1.0   
1                          0.0                      1.0   
2                          0.0                      0.0   
3                          1.0                      0.0   
4                          1.0                      0.0   

   behavioral_touch_face  ...             income_poverty  marital_status  \
0                    1.0  ...              Below Poverty     Not Married   
1                    1.0  ...              Below Poverty     Not Married   
2                    0.0  ...  <= $75,000, Above Poverty     Not Married   
3                    0.0  ...              Below Poverty     Not Married   
4                    1.0  ...  <= $75,000, Above Poverty         Married   

   rent_or_own   employment_status  hhs_geo_region                census_msa  \
0          Own  Not in Labor Force        oxchjgsf                   Non-MSA   
1         Rent            Employed        bhuqouqj  MSA, Not Principle  City   
2          Own            Employed        qufhixun  MSA, Not Principle  City   
3         Rent  Not in Labor Force        lrircsnp       MSA, Principle City   
4          Own            Employed        qufhixun  MSA, Not Principle  City   

   household_adults  household_children  employment_industry  \
0               0.0                 0.0                  NaN   
1               0.0                 0.0             pxcmvdjn   
2               2.0                 0.0             rucpziij   
3               0.0                 0.0                  NaN   
4               1.0                 0.0             wxleyezf   

   employment_occupation  
0                    NaN  
1               xgwztkwe  
2               xtkaffoo  
3                    NaN  
4               emcorrxb  

[5 rows x 36 columns]

In [4]:
training_labels.head()

respondent_id  h1n1_vaccine  seasonal_vaccine
0              0             0                 0
1              1             0                 1
2              2             0                 0
3              3             0                 1
4              4             0                 0

In [5]:
test_features.head()

respondent_id  h1n1_concern  h1n1_knowledge  behavioral_antiviral_meds  \
0          26707           2.0             2.0                        0.0   
1          26708           1.0             1.0                        0.0   
2          26709           2.0             2.0                        0.0   
3          26710           1.0             1.0                        0.0   
4          26711           3.0             1.0                        1.0   

   behavioral_avoidance  behavioral_face_mask  behavioral_wash_hands  \
0                   1.0                   0.0                    1.0   
1                   0.0                   0.0                    0.0   
2                   0.0                   1.0                    1.0   
3                   0.0                   0.0                    0.0   
4                   1.0                   0.0                    1.0   

   behavioral_large_gatherings  behavioral_outside_home  \
0                          1.0                      0.0   
1                          0.0                      0.0   
2                          1.0                      1.0   
3                          0.0                      0.0   
4                          1.0                      1.0   

   behavioral_touch_face  ...             income_poverty  marital_status  \
0                    1.0  ...                  > $75,000     Not Married   
1                    0.0  ...              Below Poverty     Not Married   
2                    1.0  ...                  > $75,000         Married   
3                    0.0  ...  <= $75,000, Above Poverty         Married   
4                    1.0  ...  <= $75,000, Above Poverty     Not Married   

   rent_or_own   employment_status  hhs_geo_region                census_msa  \
0         Rent            Employed        mlyzmhmf  MSA, Not Principle  City   
1         Rent            Employed        bhuqouqj                   Non-MSA   
2          Own            Employed        lrircsnp                   Non-MSA   
3          Own  Not in Labor Force        lrircsnp  MSA, Not Principle  City   
4          Own            Employed        lzgpxyit                   Non-MSA   

   household_adults  household_children  employment_industry  \
0               1.0                 0.0             atmlpfrs   
1               3.0                 0.0             atmlpfrs   
2               1.0                 0.0             nduyfdeo   
3               1.0                 0.0                  NaN   
4               0.0                 1.0             fcxhlnwr   

   employment_occupation  
0               hfxkjkmi  
1               xqwwgdyp  
2               pvmttkik  
3                    NaN  
4               mxkfnird  

[5 rows x 36 columns]

> - The training dataset is conveyed by 36 columns features and 3 columns labels.<br>
> - The test dataset is conveyed by 36 columns of features and we need to predict 2 columns labels.
> - In fact the first column of the different labels is the identity of the respondent, so we are not predicting it.


## 3.1 Explore the number of rows & columns, ranges of values etc.


### 3.1.1. number of rows & Columns

In [6]:
training_labels.shape

(26707, 3)

In [7]:
training_features.shape

(26707, 36)

In [8]:
training_features.shape

(26707, 36)

### 3.1.2. ranges of values

In [9]:
training_labels.min(), training_labels.max()

(respondent_id       0
 h1n1_vaccine        0
 seasonal_vaccine    0
 dtype: int64,
 respondent_id       26706
 h1n1_vaccine            1
 seasonal_vaccine        1
 dtype: int64)

>- We have 2 labels : h1n1_vaccine and seasonal_vaccine. each one has 26707 values <br>
    - 1 means the respondent took the vaccine
    - 0 means that he/she didn't took the vaccine
>- Those labels are considered as categorical value then later in visualisation, the count of value seems to be interesting to plot


In [10]:
pd.DataFrame([training_features.min(), training_features.max()], index=["min","max"])

respondent_id  h1n1_concern  h1n1_knowledge  behavioral_antiviral_meds  \
min              0           0.0             0.0                        0.0   
max          26706           3.0             2.0                        1.0   

     behavioral_avoidance  behavioral_face_mask  behavioral_wash_hands  \
min                   0.0                   0.0                    0.0   
max                   1.0                   1.0                    1.0   

     behavioral_large_gatherings  behavioral_outside_home  \
min                          0.0                      0.0   
max                          1.0                      1.0   

     behavioral_touch_face  ...  opinion_seas_vacc_effective  \
min                    0.0  ...                          1.0   
max                    1.0  ...                          5.0   

     opinion_seas_risk  opinion_seas_sick_from_vacc      age_group   race  \
min                1.0                          1.0  18 - 34 Years  Black   
max                5.0                          5.0      65+ Years  White   

        sex  hhs_geo_region                census_msa  household_adults  \
min  Female        atmpeygn  MSA, Not Principle  City               0.0   
max    Male        qufhixun                   Non-MSA               3.0   

     household_children  
min                 0.0  
max                 3.0  

[2 rows x 29 columns]

In [11]:
pd.DataFrame([test_features.min(), test_features.max()], index=["min","max"])

respondent_id  h1n1_concern  h1n1_knowledge  behavioral_antiviral_meds  \
min          26707           0.0             0.0                        0.0   
max          53414           3.0             2.0                        1.0   

     behavioral_avoidance  behavioral_face_mask  behavioral_wash_hands  \
min                   0.0                   0.0                    0.0   
max                   1.0                   1.0                    1.0   

     behavioral_large_gatherings  behavioral_outside_home  \
min                          0.0                      0.0   
max                          1.0                      1.0   

     behavioral_touch_face  ...  opinion_seas_vacc_effective  \
min                    0.0  ...                          1.0   
max                    1.0  ...                          5.0   

     opinion_seas_risk  opinion_seas_sick_from_vacc      age_group   race  \
min                1.0                          1.0  18 - 34 Years  Black   
max                5.0                          5.0      65+ Years  White   

        sex  hhs_geo_region                census_msa  household_adults  \
min  Female        atmpeygn  MSA, Not Principle  City               0.0   
max    Male        qufhixun                   Non-MSA               3.0   

     household_children  
min                 0.0  
max                 3.0  

[2 rows x 29 columns]

> - it seems that the test dataset and the training dataset was united because the first id of respondent_id in test is the last respondant_id in features + 1

> - We will reunite them in order to perform the same operation before the analysis.

In [12]:
features = pd.concat([training_features, test_features])

In [13]:
pd.DataFrame([features.min(), features.max()], index=["min", "max"])

respondent_id  h1n1_concern  h1n1_knowledge  behavioral_antiviral_meds  \
min              0           0.0             0.0                        0.0   
max          53414           3.0             2.0                        1.0   

     behavioral_avoidance  behavioral_face_mask  behavioral_wash_hands  \
min                   0.0                   0.0                    0.0   
max                   1.0                   1.0                    1.0   

     behavioral_large_gatherings  behavioral_outside_home  \
min                          0.0                      0.0   
max                          1.0                      1.0   

     behavioral_touch_face  ...  opinion_seas_vacc_effective  \
min                    0.0  ...                          1.0   
max                    1.0  ...                          5.0   

     opinion_seas_risk  opinion_seas_sick_from_vacc      age_group   race  \
min                1.0                          1.0  18 - 34 Years  Black   
max                5.0                          5.0      65+ Years  White   

        sex  hhs_geo_region                census_msa  household_adults  \
min  Female        atmpeygn  MSA, Not Principle  City               0.0   
max    Male        qufhixun                   Non-MSA               3.0   

     household_children  
min                 0.0  
max                 3.0  

[2 rows x 29 columns]

### 3.2.3. Informations on each records

In [14]:
features.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 53415 entries, 0 to 26707
Data columns (total 36 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   respondent_id                53415 non-null  int64  
 1   h1n1_concern                 53238 non-null  float64
 2   h1n1_knowledge               53177 non-null  float64
 3   behavioral_antiviral_meds    53265 non-null  float64
 4   behavioral_avoidance         52994 non-null  float64
 5   behavioral_face_mask         53377 non-null  float64
 6   behavioral_wash_hands        53333 non-null  float64
 7   behavioral_large_gatherings  53256 non-null  float64
 8   behavioral_outside_home      53251 non-null  float64
 9   behavioral_touch_face        53159 non-null  float64
 10  doctor_recc_h1n1             49095 non-null  float64
 11  doctor_recc_seasonal         49095 non-null  float64
 12  chronic_med_condition        51512 non-null  float64
 13  child_under_6_mo

> - they are some missing values to handle !
> -  it seems even though some value are binary (0 or 1) , they are stored as floats
> - Some other Characteristic are String (Object) 
> - Let us split Categorical columns from numerical columns

In [15]:
categorical_col_features = features.select_dtypes(exclude = "number").columns.tolist()
categorical_col_features

['age_group',
 'education',
 'race',
 'sex',
 'income_poverty',
 'marital_status',
 'rent_or_own',
 'employment_status',
 'hhs_geo_region',
 'census_msa',
 'employment_industry',
 'employment_occupation']

In [16]:
numerical_col_features = features.select_dtypes(include = "number").columns.tolist()
numerical_col_features

['respondent_id',
 'h1n1_concern',
 'h1n1_knowledge',
 'behavioral_antiviral_meds',
 'behavioral_avoidance',
 'behavioral_face_mask',
 'behavioral_wash_hands',
 'behavioral_large_gatherings',
 'behavioral_outside_home',
 'behavioral_touch_face',
 'doctor_recc_h1n1',
 'doctor_recc_seasonal',
 'chronic_med_condition',
 'child_under_6_months',
 'health_worker',
 'health_insurance',
 'opinion_h1n1_vacc_effective',
 'opinion_h1n1_risk',
 'opinion_h1n1_sick_from_vacc',
 'opinion_seas_vacc_effective',
 'opinion_seas_risk',
 'opinion_seas_sick_from_vacc',
 'household_adults',
 'household_children']

In [17]:
# and the labels are 
labels = training_labels.columns.tolist()[1:]
labels

['h1n1_vaccine', 'seasonal_vaccine']

# 3.2.4. Handle missing, incorrect and invalid data


#### a. Numerical columns

In [18]:
features_are_numerical = features[numerical_col_features]
features_are_numerical.head()

respondent_id  h1n1_concern  h1n1_knowledge  behavioral_antiviral_meds  \
0              0           1.0             0.0                        0.0   
1              1           3.0             2.0                        0.0   
2              2           1.0             1.0                        0.0   
3              3           1.0             1.0                        0.0   
4              4           2.0             1.0                        0.0   

   behavioral_avoidance  behavioral_face_mask  behavioral_wash_hands  \
0                   0.0                   0.0                    0.0   
1                   1.0                   0.0                    1.0   
2                   1.0                   0.0                    0.0   
3                   1.0                   0.0                    1.0   
4                   1.0                   0.0                    1.0   

   behavioral_large_gatherings  behavioral_outside_home  \
0                          0.0                      1.0   
1                          0.0                      1.0   
2                          0.0                      0.0   
3                          1.0                      0.0   
4                          1.0                      0.0   

   behavioral_touch_face  ...  health_worker  health_insurance  \
0                    1.0  ...            0.0               1.0   
1                    1.0  ...            0.0               1.0   
2                    0.0  ...            0.0               NaN   
3                    0.0  ...            0.0               NaN   
4                    1.0  ...            0.0               NaN   

   opinion_h1n1_vacc_effective  opinion_h1n1_risk  \
0                          3.0                1.0   
1                          5.0                4.0   
2                          3.0                1.0   
3                          3.0                3.0   
4                          3.0                3.0   

   opinion_h1n1_sick_from_vacc  opinion_seas_vacc_effective  \
0                          2.0                          2.0   
1                          4.0                          4.0   
2                          1.0                          4.0   
3                          5.0                          5.0   
4                          2.0                          3.0   

   opinion_seas_risk  opinion_seas_sick_from_vacc  household_adults  \
0                1.0                          2.0               0.0   
1                2.0                          4.0               0.0   
2                1.0                          2.0               2.0   
3                4.0                          1.0               0.0   
4                1.0                          4.0               1.0   

   household_children  
0                 0.0  
1                 0.0  
2                 0.0  
3                 0.0  
4                 0.0  

[5 rows x 24 columns]

In [19]:
features_are_numerical_and_null = features_are_numerical.isnull().sum()
features_are_numerical_and_null.sort_values()

respondent_id                      0
behavioral_face_mask              38
behavioral_wash_hands             82
behavioral_antiviral_meds        150
behavioral_large_gatherings      159
behavioral_outside_home          164
h1n1_concern                     177
h1n1_knowledge                   238
behavioral_touch_face            256
behavioral_avoidance             421
household_adults                 474
household_children               474
opinion_h1n1_risk                768
opinion_h1n1_sick_from_vacc      770
opinion_h1n1_vacc_effective      789
opinion_seas_vacc_effective      914
opinion_seas_risk               1013
opinion_seas_sick_from_vacc     1058
health_worker                   1593
child_under_6_months            1633
chronic_med_condition           1903
doctor_recc_h1n1                4320
doctor_recc_seasonal            4320
health_insurance               24502
dtype: int64

In [20]:
training_features.health_insurance.isnull().sum()

12274

> - The highest missing value is the "health_insurance" who has approximatively half total number of missing values.
> - Unfortunately "health_insurance" could be a factor of taking a vaccine or not.
> - **We are going to delete other records with null value except the health insurance that we should fill the missing value with an appropriate respondent answer.** <br>


>    - Firstly health worker most of the time have health_insurance we will fill health insurance values of these people with 1

In [21]:
# let us count the number of health_worker without health insurance
(features[features.health_worker == 1]["health_insurance"]== 0).value_counts()

False    5526
True      263
Name: health_insurance, dtype: int64

> Indeed most health_worker have an insurance. So let us proceed with the filling of their health_insurance

In [31]:
features_health_worker = features_are_numerical.health_worker == 1
features_are_numerical[features_health_worker]["health_insurance"].fillna(1)

41       1.0
60       1.0
77       1.0
82       1.0
85       1.0
        ... 
26649    1.0
26667    1.0
26690    1.0
26701    1.0
26704    0.0
Name: health_insurance, Length: 3089, dtype: float64

In [33]:
features_are_numerical.health_insurance.isnull().sum()

0

In [24]:
features[features.health_worker == 1]["household_children"].isnull().sum()

5

In [25]:
# first deleted respondant with missing numerical values
features_are_numerical = features_are_numerical.dropna()
features[numerical_col_features]["health_insurance"].describe()

count    28913.000000
mean         0.883824
std          0.320442
min          0.000000
25%          1.000000
50%          1.000000
75%          1.000000
max          1.000000
Name: health_insurance, dtype: float64

In [26]:
training_features.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26707 entries, 0 to 26706
Data columns (total 36 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   respondent_id                26707 non-null  int64  
 1   h1n1_concern                 26615 non-null  float64
 2   h1n1_knowledge               26591 non-null  float64
 3   behavioral_antiviral_meds    26636 non-null  float64
 4   behavioral_avoidance         26499 non-null  float64
 5   behavioral_face_mask         26688 non-null  float64
 6   behavioral_wash_hands        26665 non-null  float64
 7   behavioral_large_gatherings  26620 non-null  float64
 8   behavioral_outside_home      26625 non-null  float64
 9   behavioral_touch_face        26579 non-null  float64
 10  doctor_recc_h1n1             24547 non-null  float64
 11  doctor_recc_seasonal         24547 non-null  float64
 12  chronic_med_condition        25736 non-null  float64
 13  child_under_6_mo

In [27]:
training_features.describe()

respondent_id  h1n1_concern  h1n1_knowledge  behavioral_antiviral_meds  \
count   26707.000000  26615.000000    26591.000000               26636.000000   
mean    13353.000000      1.618486        1.262532                   0.048844   
std      7709.791156      0.910311        0.618149                   0.215545   
min         0.000000      0.000000        0.000000                   0.000000   
25%      6676.500000      1.000000        1.000000                   0.000000   
50%     13353.000000      2.000000        1.000000                   0.000000   
75%     20029.500000      2.000000        2.000000                   0.000000   
max     26706.000000      3.000000        2.000000                   1.000000   

       behavioral_avoidance  behavioral_face_mask  behavioral_wash_hands  \
count          26499.000000          26688.000000           26665.000000   
mean               0.725612              0.068982               0.825614   
std                0.446214              0.253429               0.379448   
min                0.000000              0.000000               0.000000   
25%                0.000000              0.000000               1.000000   
50%                1.000000              0.000000               1.000000   
75%                1.000000              0.000000               1.000000   
max                1.000000              1.000000               1.000000   

       behavioral_large_gatherings  behavioral_outside_home  \
count                  26620.00000             26625.000000   
mean                       0.35864                 0.337315   
std                        0.47961                 0.472802   
min                        0.00000                 0.000000   
25%                        0.00000                 0.000000   
50%                        0.00000                 0.000000   
75%                        1.00000                 1.000000   
max                        1.00000                 1.000000   

       behavioral_touch_face  ...  health_worker  health_insurance  \
count           26579.000000  ...   25903.000000       14433.00000   
mean                0.677264  ...       0.111918           0.87972   
std                 0.467531  ...       0.315271           0.32530   
min                 0.000000  ...       0.000000           0.00000   
25%                 0.000000  ...       0.000000           1.00000   
50%                 1.000000  ...       0.000000           1.00000   
75%                 1.000000  ...       0.000000           1.00000   
max                 1.000000  ...       1.000000           1.00000   

       opinion_h1n1_vacc_effective  opinion_h1n1_risk  \
count                 26316.000000       26319.000000   
mean                      3.850623           2.342566   
std                       1.007436           1.285539   
min                       1.000000           1.000000   
25%                       3.000000           1.000000   
50%                       4.000000           2.000000   
75%                       5.000000           4.000000   
max                       5.000000           5.000000   

       opinion_h1n1_sick_from_vacc  opinion_seas_vacc_effective  \
count                 26312.000000                 26245.000000   
mean                      2.357670                     4.025986   
std                       1.362766                     1.086565   
min                       1.000000                     1.000000   
25%                       1.000000                     4.000000   
50%                       2.000000                     4.000000   
75%                       4.000000                     5.000000   
max                       5.000000                     5.000000   

       opinion_seas_risk  opinion_seas_sick_from_vacc  household_adults  \
count       26193.000000                 26170.000000      26458.000000   
mean            2.719162                     2.118112          0.886499   
std             1.385055                     1.332

In [28]:
pd.options.display.max_columns = 50

In [29]:
training_features.describe()

respondent_id  h1n1_concern  h1n1_knowledge  behavioral_antiviral_meds  \
count   26707.000000  26615.000000    26591.000000               26636.000000   
mean    13353.000000      1.618486        1.262532                   0.048844   
std      7709.791156      0.910311        0.618149                   0.215545   
min         0.000000      0.000000        0.000000                   0.000000   
25%      6676.500000      1.000000        1.000000                   0.000000   
50%     13353.000000      2.000000        1.000000                   0.000000   
75%     20029.500000      2.000000        2.000000                   0.000000   
max     26706.000000      3.000000        2.000000                   1.000000   

       behavioral_avoidance  behavioral_face_mask  behavioral_wash_hands  \
count          26499.000000          26688.000000           26665.000000   
mean               0.725612              0.068982               0.825614   
std                0.446214              0.253429               0.379448   
min                0.000000              0.000000               0.000000   
25%                0.000000              0.000000               1.000000   
50%                1.000000              0.000000               1.000000   
75%                1.000000              0.000000               1.000000   
max                1.000000              1.000000               1.000000   

       behavioral_large_gatherings  behavioral_outside_home  \
count                  26620.00000             26625.000000   
mean                       0.35864                 0.337315   
std                        0.47961                 0.472802   
min                        0.00000                 0.000000   
25%                        0.00000                 0.000000   
50%                        0.00000                 0.000000   
75%                        1.00000                 1.000000   
max                        1.00000                 1.000000   

       behavioral_touch_face  doctor_recc_h1n1  doctor_recc_seasonal  \
count           26579.000000      24547.000000          24547.000000   
mean                0.677264          0.220312              0.329735   
std                 0.467531          0.414466              0.470126   
min                 0.000000          0.000000              0.000000   
25%                 0.000000          0.000000              0.000000   
50%                 1.000000          0.000000              0.000000   
75%                 1.000000          0.000000              1.000000   
max                 1.000000          1.000000              1.000000   

       chronic_med_condition  child_under_6_months  health_worker  \
count           25736.000000          25887.000000   25903.000000   
mean                0.283261              0.082590       0.111918   
std                 0.450591              0.275266       0.315271   
min                 0.000000              0.000000       0.000000   
25%                 0.000000              0.000000       0.000000   
50%                 0.000000              0.000000       0.000000   
75%                 1.000000              0.000000       0.000000   
max                 1.000000              1.000000       1.000000   

       health_insurance  opinion_h1n1_vacc_effective  opinion_h1n1_risk  \
count       14433.00000                 26316.000000       26319.000000   
mean            0.87972                     3.850623           2.342566   
std             0.32530                     1.007436           1.285539   
min             0.00000                     1.000000           1.000000   
25%             1.00000                     3.000000           1.000000   
50%             1.00000                     4.000000           2.000000   
75%             1.00000                     5.000000           4.000000   
max             1.00000                     5.000000           5.000000   

       opinion_h1n1_sick_from_vacc  opinion_seas_vacc_effective  \
count     

# Perform any additional steps (parsing dates, creating additional columns, merging multiple dataset etc.)